src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -855,69 +855,77 @@ public void remove_rollback_directory_if_exists(string packageName)
+
             var outdatedPackages = new ConcurrentDictionary<string, PackageResult>();
 
             set_package_names_if_all_is_specified(config, () => { config.IgnoreDependencies = true; });
-            var packageNames = config.PackageNames.Split(new[] { ApplicationParameters.PackageNamesSeparator }, StringSplitOptions.RemoveEmptyEntries).or_empty_list_if_null().ToList();
-
-            var repository = packageManager.SourceRepository;
-
-            bool isRemote = repository_is_remote(repository);
+            var packageNames = config.PackageNames.Split(new[] { ApplicationParameters.PackageNamesSeparator }, StringSplitOptions.RemoveEmptyEntries).or_empty_list_if_null();
 
             foreach (var packageName in packageNames)
             {
-                IPackage latestPackage;
+                var installedPackage = packageManager.LocalRepository.FindPackage(packageName);
+
+                var pkgInfo = _packageInfoService.get_package_information(installedPackage);
+                bool isPinned = pkgInfo.IsPinned;
+
+                if (config.OutdatedCommand.IgnorePinned && isPinned)
+                {
+                    string pinnedLogMessage = "{0} is pinned. Skipping pinned package.".format_with(packageName);
+                    var pinnedPackageResult = outdatedPackages.GetOrAdd(packageName, new PackageResult(installedPackage, _fileSystem.combine_paths(ApplicationParameters.PackagesLocation, installedPackage.Id)));
+                    pinnedPackageResult.Messages.Add(new ResultMessage(ResultType.Debug, pinnedLogMessage));
+                    pinnedPackageResult.Messages.Add(new ResultMessage(ResultType.Inconclusive, pinnedLogMessage));
+
+                    continue;
+                }
+
+                IQueryable<IPackage> results = repository.GetPackages().Where(x => x.Id == packageName);
 
-                if (isRemote)
+                if (config.Prerelease && repository.SupportsPrereleasePackages)
                 {
-                    latestPackage = repository.GetPackages().Where(x => x.Id == packageName && x.IsLatestVersion).ToList().SingleOrDefault();
+                    results = results.Where(p => p.IsAbsoluteLatestVersion);
                 }
                 else
                 {
-                    latestPackage = repository.FindPackage(packageName);
+                    results = results.Where(p => p.IsLatestVersion);
                 }
 
+                if (!isRemote)
+                {
+                    results = results
+                        .Where(PackageExtensions.IsListed)
+                        .Where(p => config.Prerelease || p.IsReleaseVersion())
+                        .distinct_last(PackageEqualityComparer.Id, PackageComparer.Version)
+                        .AsQueryable();
+                }
+
+                // Call .ToList() to prevent warning from NuGet if package is not found in a remote repository
+                var latestPackage = results.ToList().SingleOrDefault();

```

Comment:
This throws exceptions, if we use multiple sources, because we get 2 packages if they exists in both sources. Should we just move ```.distinct_last(PackageEqualityComparer.Id, PackageComparer.Version)``` outside ```if (!isRemote)``` and take package with the highest version?

Reply:
@DamianMaslanka5 that's a good observation. Are you on Gitter? https://gitter.im/chocolatey/choco - we could discuss in more realtime.